tests/util/dict_factory_test.py

```diff
@@ -0,0 +1,24 @@
+import sqlite3
+import pytest
+from unittest.mock import MagicMock
+from backend.util import dict_factory
+
+
+@pytest.fixture
+def mock_database_connection():
+    cursor = MagicMock(spec=sqlite3.Cursor)
+    row = MagicMock(spec=sqlite3.Row)
+
+    cursor.description = [("id", None), ("name", None), ("created_date", None)]
+
+    return cursor, row
+
+
+def test_dict_factory(mock_database_connection):
```

Comment:
might be worth having a test with an actual db spun up. could be in memory

Reply:
I think for unit tests we should keep them lean and just ensure that our code calls sqlite as expected.

We can implement E2E tests later.